akka-http2-support/src/main/scala/akka/http/impl/engine/http2/PriorKnowledgeSwitch.scala

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright (C) 2009-2019 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.http.impl.engine.http2
+
+import javax.net.ssl.SSLException
+
+import akka.util.ByteString
+import akka.NotUsed
+import akka.annotation.InternalApi
+import akka.http.impl.engine.server.HttpAttributes
+import akka.http.scaladsl.model.{ HttpRequest, HttpResponse }
+import akka.stream.TLSProtocol.{ SessionBytes, SessionTruncated, SslTlsInbound, SslTlsOutbound }
+import akka.stream.scaladsl.Flow
+import akka.stream.stage.{ GraphStage, GraphStageLogic, InHandler, OutHandler }
+import akka.stream._
+
+/** INTERNAL API */
+@InternalApi
+private[http] object PriorKnowledgeSwitch {
+  type HttpServerFlow = Flow[ByteString, ByteString, NotUsed]
+  type HttpServerShape = FlowShape[ByteString, ByteString]
+
+  private final val PRIOR_KNOWLEDGE_PREFACE = ByteString("PRI * HTTP/2.0\r\n\r\nSM\r\n\r\n")
```

Comment:
perhaps just call it `HTTP2_CONNECTION_PREFACE`, since it is sent in each HTTP2 connection

Reply:
:+1: